### PR TITLE
Example configuration to run longranger via LSF

### PR DESCRIPTION
**Pull request by muffato**
This PR is to allow running Lonranger in LSF mode and on Singularity, without the need to install longranger beforehand.

Nextflow will use our container image from GitLab, and pass on the necessary Singularity options to enable LSF.
All it needs is a couple of extra options in the pipeline's configuration file.

I'm also fixing some minor things such as MD5 checksums, the command line to get the version, etc

### CODE REVIEW

**Comment by muffato** [comment]
Confirmation the test works. In this environment, I don't have any module loaded, and longranger is not available directly.
```
$ env TMPDIR=~/nfs/scratch123/nextflow PROFILE=singularity pytest --symlink --git-aware --keep-workflow-wd-on-fail --tag longranger/align
====================================================================== test session starts ======================================================================
platform linux -- Python 3.9.5, pytest-7.0.1, pluggy-1.0.0
rootdir: /nfs/users/nfs_m/mm49/workspace/tol-it/nextflow/modules, configfile: pytest.ini
plugins: workflow-1.6.0
collecting ... 
collected 829 items                                                                                                                                             

longranger align:
        command:   nextflow run ./tests/modules/longranger/align -entry test_longranger_align -c ./tests/config/nextflow.config -c ./tests/modules/longranger/align/nextflow.config
        directory: /nfs/users/nfs_m/mm49/nfs/scratch123/nextflow/pytest_workflow_ppe4i8tu/longranger_align
        stdout:    /nfs/users/nfs_m/mm49/nfs/scratch123/nextflow/pytest_workflow_ppe4i8tu/longranger_align/log.out
        stderr:    /nfs/users/nfs_m/mm49/nfs/scratch123/nextflow/pytest_workflow_ppe4i8tu/longranger_align/log.err
'longranger align' done.

tests/test_versions_yml.py ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 15%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 33%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 52%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 70%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 89%]
sssssssssssssssssssssssssssssssssssssssssssssssssss.ssssssssssssssssssssssssssssssss                                                                      [ 99%]
tests/modules/longranger/align/test.yml .......                                                                                                           [100%]All tests succeeded. Removing temporary directories and logs. Use '--kwd' or '--keep-workflow-wd' to disable this behaviour.


========================================================== 8 passed, 821 skipped in 292.84s (0:04:52) ===========================================================
```

File: modules/longranger/align/main.nf [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+
     if (params.enable_conda) {
-        exit 1, "Conda environments cannot be used when using longranger"
-    }
-    if ( workflow.containerEngine == 'singularity' || \
-            workflow.containerEngine == 'docker' ) {
-        exit 1, "Longranger can not be run in container environment"
+        exit 1, "Conda environments cannot be used when using longranger. Please use docker or singularity containers."
     }
+    container "gitlab-registry.internal.sanger.ac.uk/tol-it/software/docker-images/longranger:${version}"
```
Comment (muffato) [inline modules/longranger/align/main.nf:13]:
```suggestion
    if ( workflow.containerEngine == 'docker' ) {
        exit 1, "Longranger is not configured to run in docker environment"
    }
    container "gitlab-registry.internal.sanger.ac.uk/tol-it/software/docker-images/longranger:${version}"
```

File: modules/longranger/align/main.nf [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
+    def version = '2.2.2-c1'
+
     if (params.enable_conda) {
-        exit 1, "Conda environments cannot be used when using longranger"
+        exit 1, "Conda environments cannot be used when using longranger. Please use docker or singularity containers."
     }
-    if ( workflow.containerEngine == 'singularity' || \
-            workflow.containerEngine == 'docker' ) {
-        exit 1, "Longranger can not be run in container environment"
+    if ( workflow.containerEngine == 'docker' ) {
```
Comment (muffato) [inline modules/longranger/align/main.nf:13]:
```suggestion
    if ( workflow.containerEngine != 'singularity' ) {
```

File: modules/longranger/align/main.nf [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
+
     if (params.enable_conda) {
-        exit 1, "Conda environments cannot be used when using longranger"
+        exit 1, "Conda environments cannot be used when using longranger. Please use docker or singularity containers."
     }
-    if ( workflow.containerEngine == 'singularity' || \
-            workflow.containerEngine == 'docker' ) {
-        exit 1, "Longranger can not be run in container environment"
+    if ( workflow.containerEngine != 'singularity' ) {
+        exit 1, "Longranger is not configured to run in docker environment"
```
Comment (muffato) [inline modules/longranger/align/main.nf:14]:
```suggestion
        exit 1, "The only container environment Longranger is configured for is docker"
```

**Review by priyanka-surana — APPROVED** [review]
The module works and incorporates singularity container appropriately with Nextflow. 👍

**Pull request merged**